app/timetables/admin.py

```diff
@@ -20,18 +25,24 @@ class AdminsInline(admin.TabularInline):
 
 
 @admin.register(Timetable)
-class TimetableAdmin(DefaultAdmin):
+class TimetableAdmin(CommonReadOnlyFieldsAdmin):
     fields = ('name', 'slug', 'code', 'api_key', 'cycle_length',
-              'current_cycle_day', 'description')
+              'current_cycle_day', 'description', 'date_created', 'date_modified')
     inlines = (AdminsInline,)
 
 
 @admin.register(Dish)
-class DishAdmin(DefaultAdmin):
-    fields = ('name', 'slug', 'description')
+class DishAdmin(CommonReadOnlyFieldsAdmin):
+    fields = ('name', 'slug', 'description', 'date_created', 'date_modified')
+
+
+@admin.register(MenuItem)
+class MenuItemAdmin(admin.ModelAdmin):
+    readonly_fields = ('date_created', 'date_modified')
+    fields = ('timetable', 'cycle_day', 'meal', 'meal_option', 'date_created', 'date_modified')
 
```

Comment:
@andela-kanyanwu: I understand the current purpose of `CommonReadOnlyFieldsAdmin` class. I would have loved it to be extendable; it seems less useful not being extendable to cater for `MenuItemAdmin` class just because of one readonly element they don't have in common. Next time another Admin class has just this `readonly_fields = ('date_created', 'date_modified')`, we will want to abstract it out into another class (`SomethingCommonReadOnlyFieldsAdmin`) and make both inherit from it, right? If another Admin class comes up with `readonly_fields = ('something_readonly', 'date_created', 'date_modified')`, we would not be able to extend `CommonReadOnlyFieldsAdmin` nor `SomethingCommonReadOnlyFieldsAdmin`.

If you can't find a way around this at the moment, we can go with the current implementation though; it's not a big deal.